### PR TITLE
Allow to provide custom settings.xml

### LINKED ISSUE #706: Setup the maven proxy inside of the builder
It could be helpful if kamel can provide an option to setup the [proxy](https://maven.apache.org/settings.html#Proxies) in the maven setting.xml. 

We could do it by overriding the base image maven setting, but it could be much handy if we can change it through kamel command.

Any thought?

### LINKED ISSUE #715: Allow to provide custom settings.xml
This supersede #706

Should we also remove repositories set-up and let the user configure the maven build only through a settings.xml ?

/cc @astefanutti

### PR DESCRIPTION
**Pull request by lburgazzoli**
So now you can configure maven by providing a custom settings.xml from a configmap/secret:

```yaml
apiVersion: camel.apache.org/v1alpha1
kind: IntegrationPlatform
metadata:
  labels:
    app: camel-k
  name: camel-k
spec:
  build:
    maven:
      settings:
        configMapKeyRef:
          key: s.xml
          name: maven-settings
```

**NOTE**: the possibility to set maven repositories has been removed from both platform and integration so the only option to configure maven is through the platform itself.

Fixes #715
Fixes #706

### CODE REVIEW

File: pkg/cmd/install.go [hunk 1]
```diff
@@ -62,8 +64,6 @@ func newCmdInstall(rootCmdOptions *RootCmdOptions) *cobra.Command {
 	cmd.Flags().StringVar(&impl.registry.Address, "registry", "", "A Docker registry that can be used to publish images")
 	cmd.Flags().StringVar(&impl.registry.Secret, "registry-secret", "", "A secret used to push/pull images to the Docker registry")
 	cmd.Flags().BoolVar(&impl.registry.Insecure, "registry-insecure", false, "Configure to configure registry access in insecure mode or not")
-	cmd.Flags().StringSliceVar(&impl.repositories, "repository", nil, "Add a maven repository")
-	cmd.Flags().BoolVar(&impl.snapshotRepositories, "snapshot-repositories", false, "Automatically include known snapshot repositories")
 	cmd.Flags().StringVar(&impl.localRepository, "local-repository", "", "Location of the local maven repository")
```
Comment (astefanutti) [inline pkg/cmd/install.go]: I wonder whether the local repository customisation should just be removed and covered with the Maven settings?

Comment (lburgazzoli) [reply pkg/cmd/install.go]: it is possible but then we need to provide our default settings.xml and eventually merge it with the one provided by the user

Comment (astefanutti) [reply pkg/cmd/install.go]: Right, we can do it later if we want it to...

File: README.adoc [hunk 2]
```diff
@@ -156,49 +156,50 @@ As example if you want to change the queue size of the seda component, you can u
 camel.component.seda.queueSize = 10
 ```
 
-==== Configure additional maven repositories
-
-Additional maven repositories can be defined at platform installation time or at integration/context build time:
+==== Configure maven settings
 
+Maven settings can be configured by providing a custom settings.xml through a CaonfigMap or secret.
```
Comment (astefanutti) [inline README.adoc]: CaonfigMap -> ConfigMap

File: README.adoc [hunk 3]
```diff
@@ -156,49 +156,50 @@ As example if you want to change the queue size of the seda component, you can u
...
 
-[NOTE]
-====
-The final repositories list is the sum of the repositories defined on the resource (integration/context) and the platform ones
-====
+To quickly create a config map you can use the following command:
+
+[source]
+----
+kubectl create configmap maven-settings --from-file=settigns.xml
```
Comment (astefanutti) [inline README.adoc]: settigns.xml -> settings.xml

File: README.adoc [hunk 4]
```diff
@@ -156,49 +156,50 @@ As example if you want to change the queue size of the seda component, you can u
 camel.component.seda.queueSize = 10
 ```
 
-==== Configure additional maven repositories
-
-Additional maven repositories can be defined at platform installation time or at integration/context build time:
+==== Configure maven settings
```
Comment (astefanutti) [inline README.adoc]: maven -> Maven

File: README.adoc [hunk 5]
```diff
@@ -156,49 +156,50 @@ As example if you want to change the queue size of the seda component, you can u
...
 
+Maven settings can be configured by providing a custom settings.xml through a CaonfigMap or secret.
+The source for maven setttings can be configuired using either the `kamel` cli at installation time or by editing Camel K's IntegrationPlatform:
```
Comment (astefanutti) [inline README.adoc]: cli -> CLI

**Comment by lburgazzoli** [comment]
> Some nitpicking :)

oh man, how may typos was I able to make in 10 lines ?

**Comment by nicolaferraro** [comment]
I love the idea of the configmap/secret to set the maven settings. But I'd prefer keeping the `--repository` flag because it's handy and I've used it in many installations/quickstarts and I wanted to use it for linking camel k to the staging repo when releasing.

I was thinking of keeping the `--repositories` as shortcut, alternative to maven settings (cannot be used together, for simplicity). Under the hood, it can even generate a simple `settings.xml` in a configmap if this simplifies code. Wdyt?

**Comment by lburgazzoli** [comment]
We can add some short-cuts on `kamel install` so the xml is automatically created, could that work ?

**Comment by nicolaferraro** [comment]
Yeah, that's what I meant ;)

**Comment by lburgazzoli** [comment]
reworking it

**Pull request closed without merging**